Controller/Callback/Offsite.php

```diff
@@ -163,6 +165,8 @@ public function execute()
         }
 
         try {
+            // adding delay to cover the delay in updating the charge status in the Omise backend
+            usleep(500000);
```

Comment:
would it be better to set for example 3 retries? then if it's still pending then return as pending. Not sure if sleep is a good idea for async, what if it takes longer than 0.5 seconds?

Reply:
This issue is random and the delay is a temporary fix until the payment response flow is changed. We tested with 0.5 secs to 1 sec timing and selected the lowest time. Retry option would be better but it would slow thing for all the payment method.

Reply:
I see, if this is a temporary fix then should be fine. Retry option would be faster though I think as we can do the first attempt immediately then retry after 0.25s and then after 0.5s. This means the longest wait time is 0.5s but it could also be resolved immediately. Either solution is ok with me since this is a temporary fix.

Reply:
Yeah, It would definitely be faster but the thing we do not know the expected charge status and all the status `success`, `pending`, and `failed` are valid. So we cannot end after first try as we won't know if the charge status we got is the actual status. So, we have to execute all the retry and use the last charge response.